src/frontend/src/components/Banner/Banner.jsx

```diff
@@ -112,87 +112,99 @@ function ScrollDown(props) {
+
+    if (errorCB) {
+      errorCB();
+    }
+  }
+}
+
+function RetrieveBannerDynamicAssets() {
   const [backgroundImgSrc, setBackgroundImgSrc] = useState('');
   const [transitionBackground, setTransitionBackground] = useState(true);
+  const [stats, setStats] = useState({ stats: { posts: 0, authors: 0, words: 0 } });
+
+  const classes = useStyles();
+  const { telescopeUrl } = useSiteMetadata();
 
   useEffect(() => {
     async function getBackgroundImgSrc() {
-      try {
-        // Uses https://unsplash.com/collections/894/earth-%26-planets collection
-        /* Other Options: 
+      // Uses https://unsplash.com/collections/894/earth-%26-planets collection
+      /* Other Options: 
         - https://unsplash.com/collections/2411320/trend%3A-extreme-neon
         - https://unsplash.com/collections/1538150/milkyway
         - https://unsplash.com/collections/291422/night-lights
         */
 
-        // Ensure we are using an image which fits correctly to user's viewspace
-        const dimensions = `${window.innerWidth}x${window.innerHeight}`;
-        const response = await fetch(`https://source.unsplash.com/collection/894/${dimensions}/`);
-
-        if (response.status !== 200) {
-          throw new Error(response.statusText);
+      // Ensure we are using an image which fits correctly to user's viewspace
+      const dimensions = `${window.innerWidth}x${window.innerHeight}`;
+
+      await getDynamicAsset(
+        `https://source.unsplash.com/collection/894/${dimensions}/`,
+        response => {
+          // Ease in Background
+          setBackgroundImgSrc(response.url);
+        },
+        () => {
+          // Fallback to default image
+          setBackgroundImgSrc('../../images/hero-banner.png');
         }
-
-        const src = response.url;
-
-        // Ease in Background
-        setBackgroundImgSrc(src);
-        setTransitionBackground(false);
-      } catch (error) {
-        console.error('Error getting user info', error);
-        // Fallback to default image
-        setBackgroundImgSrc('../../images/hero-banner.png');
-      }
+      );
     }
 
-    getBackgroundImgSrc();
-  }, []);
-
-  return (
-    <div
-      className="bannerImg"
-      style={{
-        backgroundImage:
-          backgroundImgSrc === '../../images/hero-banner.png'
-            ? backgroundImgSrc
-            : `url(${backgroundImgSrc})`,
-        opacity: transitionBackground ? 0 : 0.4,
-      }}
-    ></div>
-  );
-}
-
-function RetrieveStats() {
-  const { telescopeUrl } = useSiteMetadata();
-  const [stats, setStats] = useState({ stats: { posts: 0, authors: 0, words: 0 } });
-
-  useEffect(() => {
     async function getStats() {
-      try {
-        const response = await fetch(`${telescopeUrl}/stats/year`);
-        if (response.status !== 200) {
-          throw new Error(response.statusText);
-        }
-
-        const stat = await response.json();
+      await getDynamicAsset(`${telescopeUrl}/stats/year`, response => {
```

Comment:
Tried it on Zeit, and I still dislike the stats line coming in before the image.  What if you initiate this in the success callback of the other call for the background?

Reply:
That did it :D